src/mongocxx/lib/mongocxx/v_noabi/mongocxx/pool.cpp

```diff
@@ -131,8 +131,12 @@ pool::entry::operator bool() const noexcept {
 pool::entry::entry(pool::entry::unique_client p) : _client(std::move(p)) {}
 
 pool::entry pool::acquire() {
-    return entry(entry::unique_client(new client(libmongoc::client_pool_pop(_impl->client_pool_t)),
-                                      [this](client* client) { _release(client); }));
+    auto cli = libmongoc::client_pool_pop(_impl->client_pool_t);
+    if (!cli)
+        throw exception{ error_code::k_invalid_client_object,
```

Comment:
Suggest defining a new error code. `k_invalid_client_object` is associated [with the message](https://github.com/mongodb/mongo-cxx-driver/blob/54693de778fffb56571853d7f22c7e972d53b845/src/mongocxx/lib/mongocxx/v_noabi/mongocxx/exception/error_code.cpp#L38) `invalid use of default constructed or moved-from mongocxx::client object`.